sky/serve/replica_managers.py

```diff
@@ -601,7 +647,15 @@ def _download_and_stream_logs(info: ReplicaInfo):
         self._down_process_pool[replica_id] = p
 
     def scale_down(self, replica_id: int) -> None:
-        self._terminate_replica(replica_id, sync_down_logs=False)
+        if replica_id in self._launch_process_pool:
+            self._terminate_replica(replica_id, sync_down_logs=False)
+            info = serve_state.get_replica_info_from_id(self._service_name,
+                                                        replica_id)
+            assert info is not None
+            info.status_property.sky_launch_status = ProcessStatus.INTERRUPTED
+            del self._launch_process_pool[replica_id]
+        else:
+            self._terminate_replica(replica_id, sync_down_logs=False)
```

Comment:
could we move this logic to `_terminate_replica`?

Reply:
I think putting it here better. 
This is for `scale_down` launching replicas. When it happens, we need graceful termination (by setting process status to be interrupted)
Other cases where `terminate_replicas` is called, we don't need to set `sky_launch_status` to be `INTERRUPTED`, etc.. For example, when there is actually an error in launching.

Reply:
This could happen when the downscale seconds are configured to a small number?

Reply:
i.e. scale down a replica that is launching

Reply:
That is also an issue faced by the multi-acc research; not sure if we should separate into another PR too.

Reply:
I see let me move it to `_terminate_replica` then